realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -2976,4 +2977,187 @@ public void clear_all() {
         assertEquals(0, realm.where(Cat.class).count());
         assertTrue(realm.isEmpty());
     }
+
+    // Test if waitForChange gets waked up by 1) empty transaction 2) transaction with some data changes.
+    @Test
+    public void waitForChange() throws InterruptedException {
```

Comment:
I'm going to remove all the duplicated tests in `DynamicRealmTests` and create another PR for unifying the RealmTests/DynamicRealmTests since this PR seems to be on hold for a while.